### DISCUSSION TITLE
New Icons in New UI

### DISCUSSION THREAD

**Discussion by bryankhor07** [google/timesketch#2928] [General]
I can't seem to find the icons in the new UI. My teammates and I are trying to add tooltips to these new icons but we can't find them. The list of icons is in here -> #2693

**Comment by jkppr** [comment]
Hi Bryan, can you elaborate on what you mean by "can't find them"?

If it helps, here is an example for an icon without tooltip:
https://github.com/google/timesketch/blob/master/timesketch/frontend-ng/src/components/Explore/EventList.vue#L83

And here is an example with a working tooltip:
https://github.com/google/timesketch/blob/master/timesketch/frontend-ng/src/components/Explore/EventDetail.vue#L92

Our default tooltip delay is at 500 at the moment. This can be changed depending on the location of the icon and it's context though.

**Reply by bryankhor07** [reply]
Hi jkppr, thanks for the clarification. I have another question. Do I need to add tooltips for the icons on the left panel?
![image](https://github.com/google/timesketch/assets/67664138/fac1503b-9d2f-45dc-a09f-55307f110b32)

**Comment by jkppr** [comment]
Not all icons need tooltips. For example the ones in the left panel already have a descriptive term next to it, so it is not necessary for the beginning.

Let's start with the icons that can be clicked (trigger an action) and have no descriptive text next to it.

Here is a good read about tooltips & UX: https://www.nngroup.com/articles/tooltip-guidelines/#:~:text=Don't%20use%20tooltips%20for,seem%20like%20an%20unusual%20request.

**Reply by bryankhor07** [reply]
Ok got it!

**Comment by bryankhor07** [comment]
@jkppr Would this be the action menu?
![image](https://github.com/google/timesketch/assets/67664138/860d8d20-e9cd-4bc5-95ba-8c702a6e9469)

**Reply by jkppr** [reply]
yes I would call most of the 3 dot icons "action menus". You can find another one left to each event entry for example.

**Reply by bryankhor07** [reply]
Cool, thanks!

**Comment by bryankhor07** [comment]
@jkppr Can I use the title property for tooltip instead of wrapping the icon with "<v - tooltip>"? Sometimes, wrapping it with "<v - tooltip>" doesn't work. I find it easier to just use the title property.

**Reply by jkppr** [reply]
Sure, as long as the result is a proper tooltip information explaining the action that will happen when clicking the icon.

**Reply by jkppr** [reply]
I looked at your PR #2952 and the tooltips created by using the title seem to be the default ones provided by the browser. They have the downside that we cannot control the delay time and the look of the tooltip. Given these circumstances I would recommend to use the title only where v-tooltip is not possible at all.

**Comment by jkppr** [comment]
@bryankhor07 feel free to send some partial PR my way for review when you have added a tooltip to some icons already. This way we prevent that there will be one big PR in the end and can include feedback early.

**Reply by bryankhor07** [reply]
Alright sounds good. I do have another question. Where is the manage icon located in the left panel folder? I can't seem to find it.
![image](https://github.com/google/timesketch/assets/67664138/320f16fa-988e-4e97-b6b7-aea2424208f6)

**Comment by jkppr** [comment]
I think you are looking for this one: https://github.com/google/timesketch/blob/master/timesketch/frontend-ng/src/views/Sketch.vue#L66

**Reply by bryankhor07** [reply]
Thanks!!

**Comment by jkppr** [comment]
Topic closed in PR https://github.com/google/timesketch/pull/2983
